### ISSUE TITLE
Port twisted.python.monkey to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5896]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5896|
|Type|enhancement|
|Created|2012-08-21 13:23:25Z|
|Branch|https://github.com/twisted/twisted/tree/monkey-python3-5896|

trial requires this module to run.

<details><summary>Searchable metadata</summary>

```
trac-id__5896 5896
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_monkey_python3_5896 branches/monkey-python3-5896
branch_author__itamar itamar
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1345555405000000 1345555405000000
changetime__1345739310000000 1345739310000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35321]) Branching to 'monkey-python3-5896'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

Once again, strings need to be either native (attributes), or are for testing purposes only and so needn't be changed (values set on attributes).

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/monkey-python3-5896

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

Oh, and there's full test coverage according to coverage.py.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @itamarst<br>@exarkun set status to `new`|
|-|-|

  1. fwiw, the coding standard does not call for 3 blank lines between imports and the first function/class definition.  It calls for 3 blank lines between _suites_.  Personally I like one or two blank lines separating imports from what follows.
  1. As long as we're adding `__future__.division` to all modules, even ones where it makes no difference, how about `__future__.absolute_import` too?

That's it.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set status to `closed`|
|-|-|

(In [35363]) Merge monkey-python3-5896: twisted.python.monkey now certified to run on Python 3.

Author: itamar
Review: exarkun
Fixes: [#5896](https://github.com/twisted/twisted/issues/5896)
